manager/controllers/app/m4dapplication_controller.go

```diff
@@ -471,15 +471,29 @@ func (r *M4DApplicationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 // AnalyzeError analyzes whether the given error is fatal, or a retrial attempt can be made.
 // Reasons for retrial can be either communication problems with external services, or kubernetes problems to perform some action on a resource.
 // A retrial is achieved by returning an error to the reconcile method
-func AnalyzeError(app *app.M4DApplication, log logr.Logger, assetID string, err error) error {
-	errStatus, _ := status.FromError(err)
-	log.V(0).Info(errStatus.Message())
-	if errStatus.Code() == codes.InvalidArgument {
-		setCondition(app, assetID, errStatus.Message(), true)
-		return nil
+func AnalyzeError(application *app.M4DApplication, assetID string, err error) {
+	if err == nil {
+		return
+	}
+	// Unwrap the error - it wraps the original error by the connector
+	errorToCheck := err
+	if cause := errors.Cause(err); cause != nil {
+		errorToCheck = cause
+	}
+	if errStatus, fromGrpc := status.FromError(errorToCheck); fromGrpc {
+		if errStatus.Code() == codes.InvalidArgument {
+			setDenyCondition(application, assetID, errStatus.Message())
```

Comment:
So there is an undocumented API here that if the asset is unknown the connector must return InvalidArgument? Do we need to document it?

Reply:
I think we do.

Reply:
There is also a leakage of the abstractions here by looking at grpc specifics. That will be an issue when we move to openapi for example. Maybe the right approach is to change the DataCatalog interface to convert to `app.*` errors.  If this works for our current connector then it's fine for this PR but we will need to resolve it later.